engine_bs/README.md

```diff
@@ -15,17 +15,53 @@ sh <(curl -sL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.
 opam init
 ```
 
-2. Cloning repositories
+2. BuckleScript
+
+Clone the BuckleScript fork that works with Sketch.sh. `bucklescript` root project folder
+should be next to `sketch-sh` root project folder in the directory structure.
 
 ```sh
-git clone https://github.com/BuckleScript/bucklescript
-git clone https://github.com/BuckleScript/bucklescript-playground
-git clone https://github.com/facebook/reason/
+git clone https://github.com/thangngoc89/bucklescript/
+git checkout sketch
+```
+
+js_of_ocaml is also needed to build the browser version of the BuckleScript compiler:
+
+```sh
+opam install js_of_ocaml
+```
+
+_Note: js_of_ocaml can be installed in any opam switch, even different than 4.02.3._
+
+```sh
+git submodule update --init && node scripts/buildocaml.js # Build BuckleScript fork of OCaml
+cd scripts
+./ninja.js config && ./ninja.js build # Build BuckleScript
+BS_PLAYGROUND=../../sketch-sh/client/public ./repl.js # Generate browser version of BuckleScript
+```
+
+If there are errors while building BuckleScript, try running `./scripts/ninja.js clean` and calling
+`./ninja.js config && ./ninja.js build` again.
+
+For convenience, the last two commands (`./ninja.js build` and `./repl.js`) are included in a script file, so when
+new changes are done in BuckleScript code, one can just run `./repl.sh` from the `scripts` folder:
+
+```sh
+cd scripts && ./repl.sh
```

Comment:
nice catch!.

Reply:
@thangngoc89 this actually depends on some changes in the bucklescript fork that I'm suggesting in https://github.com/thangngoc89/bucklescript/pull/1.